pulsar/consumer_multitopic.go

```diff
@@ -189,25 +189,30 @@ func ackIDListFromMultiTopics(log log.Logger, msgIDs []MessageID, findConsumer f
 		}
 	}
 
-	ackError := AckError{}
-	for consumer, ids := range consumerToMsgIDs {
-		if err := consumer.AckIDList(ids); err != nil {
-			if topicAckError := err.(AckError); topicAckError != nil {
+	subErrCh := make(chan error, len(consumerToMsgIDs))
+	errCh := make(chan error)
+	go func() {
```

Comment:
Yes. I used to use the goroutine to support timeout mechanism, e.g.

```go
	select {
	case <-time.After(5 * time.Second):
		return errors.New("timeout waiting for ackIDList")
	case <-errCh:
		return <-errCh
	}
```

But it seems unnecessarily complicated because each `subErrCh` has its own timeout. So it's now unnecessary